oauth.go

```diff
@@ -262,54 +273,97 @@ func (svc *AlbyOAuthService) AppsShowHandler(c echo.Context) error {
 func (svc *AlbyOAuthService) AppsNewHandler(c echo.Context) error {
 	sess, _ := session.Get("alby_nostr_wallet_connect", c)
 	userID := sess.Values["user_id"]
-	appName := c.QueryParam("c") // c - for client
+	name := c.QueryParam("c") // c - for client
 	if userID == nil {
-		return c.Redirect(302, "/?c="+appName)
+		return c.Redirect(302, "/?c="+name)
+	}
+
+	if name != "" {
```

Comment:
hmm, this is tricky because it is not a good practice to do changes on a GET request. For example if I now share a link nwc.getalby.com/apps/new?c=bar then it will automatically create a session. or if I reload the page it will create another session.
Back in the day this could even be cross-side scripted easily

if we want to do this here, then I'd rely on the session here and then remove the session here. 

but I'd say the risk of side effects here is too big and giving the user button to confirm is not that bad. (there are other things to improve first)

Reply:
I disagree that the risk is too big and we need to remove hoops that the user has to jump through.
I agree that we should depend on the cookie session and clear it when we created something though.

Reply:
but then let's check the session in here. 
I find it strange if I click on the link in Amethyst and a session is created without me confirming it. 
or if I send you a link and it will automatically create a session for you when you click on it. 

Also browser or cloudflare might cache it, etc. GET requests are normally not for writing data - there are reasons why HTTP - REST is the way it is speced.